hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/FederationAccessManagerBase1.java

```diff
@@ -194,16 +193,19 @@ else if ( countValueObj instanceof Long value ) {
 	protected static class FunctionToObtainCardinalityResponseFromTPFResponse implements Function<TPFResponse, CardinalityResponse>
 	{
 		public CardinalityResponse apply( final TPFResponse tpfResp ) {
-			if ( tpfResp == null ) throw new IllegalArgumentException("The given TPFResponse is null");
+			if ( tpfResp == null ) {
+				throw new IllegalArgumentException( "The given TPFResponse is null" );
+			}
 
-			final Integer cardinality = tpfResp.getCardinalityEstimate();
-			if ( cardinality != null ) {
-				final int c = cardinality;
-				return new CardinalityResponseImpl(tpfResp, tpfResp.getRequest(), c);
+			final Integer cardinality;
+			try {
+				cardinality = tpfResp.getCardinalityEstimate();
 			}
-			else {
-				return new CardinalityResponseImplWithoutCardinality( tpfResp, tpfResp.getRequest() );
+			catch ( final CardinalityEstimationUnavailableError e ) {
```

Comment:
Related to my other review comment, this is the point where you should create `CardinalityEstimationUnavailableError` instead of catching it. In other words, revert this part here back to the if-else version that we have currently (before this PR) and, then, create `CardinalityEstimationUnavailableError` within the else-branch.